tools/packaging/helm-chart/kata-deploy/templates/kata-gpu-feature-discovery.yaml

```diff
@@ -0,0 +1,75 @@
+{{- if .Values.kataGpuFeatureDiscovery -}}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: kata-gpu-feature-discovery
+  namespace: {{ .Release.Namespace }}
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: kata-gpu-feature-discovery
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: kata-gpu-feature-discovery
+    spec:
+      runtimeClassName: kata-qemu-nvidia-gpu
+      nodeSelector:
+        katacontainers.io/gpu-feature-discovery: required
+{{- with .Values.kataDeploy.imagePullSecrets }}
+      imagePullSecrets:
+{{- toYaml . | nindent 6 }}
+{{- end }}
+      initContainers:
+        - image: nvcr.io/nvidia/k8s-device-plugin:v0.17.0
+          name: gpu-feature-discovery
+          command: ["/usr/bin/gpu-feature-discovery"]
+          resources:
+            requests:
+              "nvidia.com/pgpu": 1
+            limits:
+              "nvidia.com/pgpu": 1
+          volumeMounts:
+            - name: output-dir
+              mountPath: "/etc/kubernetes/node-feature-discovery/features.d"
+            - name: host-sys
+              mountPath: "/sys"
+          securityContext:
+            privileged: true
```

Comment:
I believe privileged may only be required for a certain subset of labels -- specifically the `nvidia.com/vgpu.host-driver-version` and `nvidia.com/vgpu.host-driver-branch` labels which are only applicable when running in a vGPU guest. https://github.com/NVIDIA/k8s-device-plugin/blob/44b263c37e13370f857ef0691b8ab8ff02135c0d/internal/lm/vgpu.go#L39-L58